nym-vpn-core/crates/nym-vpn-lib/src/bandwidth_controller.rs

```diff
@@ -80,6 +95,81 @@ impl SpecificGatewayError {
                 | SpecificGatewayError::TopUpWireguard { .. }
         )
     }
+
+    pub fn from_deprecated_topup_wireguard(
+        gateway_id: String,
+        ticketbook_type: TicketType,
+        source: nym_authenticator_client::AuthenticationClientError,
+    ) -> Self {
+        if matches!(
+            source,
+            nym_authenticator_client::AuthenticationClientError::TimeoutWaitingForConnectResponse
+        ) {
+            return SpecificGatewayError::GatewayTimeout { gateway_id };
+        }
+
+        SpecificGatewayError::DeprecatedTopUpWireguard {
+            gateway_id,
+            ticketbook_type,
+            source: Box::new(source),
+        }
+    }
+
+    pub fn from_topup_wireguard(
+        gateway_id: String,
+        ticketbook_type: TicketType,
+        source: MetadataClientError,
+    ) -> Self {
+        if let MetadataClientError::HttpClientError(e) = &source
```

Comment:
For the metadata client queries, we are using simple HTTP requests so we shouldn't bother with implementing retries at this level. Instead look at where we build the HTTP client and configure it with retries from the builder, using the builder's `with_retries` method, in:
```
let inner = nym_http_api_client::Client::builder(base_url)
            .and_then(|builder| builder.with_reqwest_builder(reqwest_builder).build())?;
```

Reply:
Done.